app/packages/registration-schema/page-manager.ts

```diff
@@ -0,0 +1,55 @@
+import EmberObject from '@ember/object';
+import Changeset from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+import { ChangesetDef } from 'ember-changeset/types';
+import {
+    buildValidation,
+    getSchemaBlockGroups,
+    ResponseValue,
+    SchemaBlock,
+    SchemaBlockGroup,
+} from 'ember-osf-web/packages/registration-schema';
+import { PageResponse } from 'ember-osf-web/packages/registration-schema/page-response';
+
+export class PageManager extends EmberObject {
+    changeset: ChangesetDef;
+    schemaBlockGroups: SchemaBlockGroup[];
+    pageResponses: Record<string, ResponseValue>;
+    pageHeadingText: string;
+    isVisited: boolean;
+
+    constructor(pageSchemaBlocks: SchemaBlock[], registrationResponses: PageResponse) {
+        super();
+        this.schemaBlockGroups = getSchemaBlockGroups(pageSchemaBlocks);
+        this.pageHeadingText = this.schemaBlockGroups[0].labelBlock!.displayText!;
+        this.pageResponses = {};
+        this.isVisited = false;
+        for (const group of this.schemaBlockGroups) {
+            const groupKey = group.registrationResponseKey;
```

Comment:
Could you please rename this `groupKey` to something close to `registrationResponseKey`? For those of us still catching up with the renaming, it's a bit confusing.

Reply:
Once this is merged, you can rebase and rename it in your PR.

Reply:
👍